plugins/inputs/gnmi/gnmi.go

```diff
@@ -55,6 +55,8 @@ type GNMI struct {
 	acc     telegraf.Accumulator
 	cancel  context.CancelFunc
 	wg      sync.WaitGroup
+	// Lookup/device/name/key/value
+	lookup map[string]map[string]map[string]map[string]interface{}
```

Comment:
this seems like an odd structure. 
Do we know nothing about the data structure? 
Does it have always exactly 4 segments? 
can we store it as a single string key (and parse it into pieces on demand) instead?

Reply:
The premise of the plugin is to treat some of the different update messages we get from the gNMI server as "data to be tagged on other, non-tag-specific updates".  IE, if there is an openconfig endpoint that can map `interface name` -> `interface description`, when we are receiving updates giving us counter information keyed by `interface name`, go ahead and tag that update with the value from the known `interface description`.  

In order to do that, we need to track and store these key/value pairs, and index them by lookup (the gNMI subscription/endpoint), which device we receive them from, and the key value for this specific lookup (eg, `interface name` in our previous example, with the bottom layer being key/value pairs for the actual tags to apply to the measurement.

We need the top-level map (string ->map) for the specific lookup, so we can perform the retrieve test in [L304](https://github.com/influxdata/telegraf/pull/7484/files#diff-8f015c9f22f0f8f4f8ee17b671e6d8f3ffa1871b86fc765a96e00d9d45ba0f39R304) when processing an update from the server, to see if it should be used to update our tag lookup map, or process it as a normal telemetry update.

The bottom layer map of tag name -> tag value is needed when enumerating them to apply to other updates in [L314](https://github.com/influxdata/telegraf/pull/7484/files#diff-8f015c9f22f0f8f4f8ee17b671e6d8f3ffa1871b86fc765a96e00d9d45ba0f39R314)

Would it make sense to collapse the middle two `map[string]map[string]` for `device` and `interface name` into an anonymous struct?

`lookup map[string]map[{device, name string}]map[string]interface{}`

Really do appreciate the feedback on this -- it was the section of the code I was least confident in.